pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -425,3 +437,19 @@ func buildTemplateMapMD(clusterSpec *cluster.Spec, workerNodeGroupMachineSpec v1
 	}
 	return values
 }
+
+func (p *tinkerbellProvider) NodeGroupsToDelete(ctx context.Context, workloadCluster *types.Cluster, currentSpec, newSpec *cluster.Spec) ([]*clusterv1.MachineDeployment, error) {
+	workerConfigs := buildMapForWorkerNodeGroupsByName(newSpec.Spec.WorkerNodeGroupConfigurations)
+	machineDeployments := make([]*clusterv1.MachineDeployment, 0, len(currentSpec.Spec.WorkerNodeGroupConfigurations))
+	for _, prevWorkerNodeGroupConfig := range currentSpec.Spec.WorkerNodeGroupConfigurations {
+		if _, ok := workerConfigs[prevWorkerNodeGroupConfig.Name]; !ok {
+			mdName := fmt.Sprintf("%s-md-0", workloadCluster.Name)
```

Comment:
Why is this `md-0`?

Reply:
Tinkerbell provider uses a `workerPoolName` to name the machineDeployment, and it is currently hard-coded to `md-0`. I need to check with the providers team to see how they want to handle naming for multiple groups, but for now it's always `md-0`

Reply:
Does that mean that multiple worker node groups is not supported for tinkerbell?

Reply:
Yes, correct. The logic is in place for it to work except for the naming in the tinkerbell `template-md.yaml`. I have to discuss with the providers team to understand how they want to go about it since the tinkerbell template has the extra `pool` field that is currently being used to name those CAPI objects

Reply:
Cool. Do we have ticket for this? It doesn't matter if it's in the tinkerbell project as long as we have one

Reply:
I have it on my personal TODO list but I'll create a ticket on here to track it